src/main/java/org/openstreetmap/atlas/utilities/filters/AtlasEntityPolygonsFilter.java

```diff
@@ -0,0 +1,520 @@
+
+/**
+ * Configurable {@link AtlasEntity} filter that passes through anything that intersects with
+ * includePolygons polygons or anything that doesn't intersect with exclude polygons. Exclude
+ * polygons are looked at only if no includePolygons polygons exist. Include takes precedence and
+ * polygons intersecting with previously read polygons are dropped with a warning.
+ * 
+ * @author jklamer
+ */
+public final class AtlasEntityPolygonsFilter implements Predicate<AtlasEntity>, Serializable
+{
+    public static final String EXCLUDED_MULTIPOLYGONS_KEY = "filter.multipolygons.exclude";
+    public static final String EXCLUDED_POLYGONS_KEY = "filter.polygons.exclude";
+    public static final String INCLUDED_MULTIPOLYGONS_KEY = "filter.multipolygons.include";
+    public static final String INCLUDED_POLYGONS_KEY = "filter.polygons.include";
+    private static final Logger logger = LoggerFactory.getLogger(AtlasEntityPolygonsFilter.class);
+    public static final IntersectionDeciding DEFAULT_INTERSECTION_DECIDING = new IntersectionDeciding()
+    {
+        @Override
+        public boolean multiPolygonEntityIntersecting(final MultiPolygon multiPolygon,
+                final AtlasEntity entity)
+        {
+            if (entity instanceof LineItem)
+            {
+                return multiPolygon.overlaps(((LineItem) entity).asPolyLine());
+            }
+            if (entity instanceof LocationItem)
+            {
+                return multiPolygon
+                        .fullyGeometricallyEncloses(((LocationItem) entity).getLocation());
+            }
+            if (entity instanceof Area)
+            {
+                return multiPolygon.overlaps(((Area) entity).asPolygon());
+            }
+            if (entity instanceof Relation)
+            {
+                return ((Relation) entity).members().stream().map(RelationMember::getEntity)
+                        .anyMatch(relationEntity -> this
+                                .multiPolygonEntityIntersecting(multiPolygon, relationEntity));
+            }
+            else
+            {
+                logger.warn("Unknown AtlasEntity Implementation {}", entity);
+                return false;
+            }
+        }
+
+        @Override
+        public boolean polygonEntityIntersecting(final Polygon polygon, final AtlasEntity entity)
+        {
+            return entity.intersects(polygon);
+        }
+    };
+    private static final long serialVersionUID = -3474748398986569205L;
+    private List<MultiPolygon> excludeMultiPolygons = new ArrayList<>();
+    private List<Polygon> excludePolygons = new ArrayList<>();
+    private Type filterType;
+    private List<MultiPolygon> includeMultiPolygons = new ArrayList<>();
+    private List<Polygon> includePolygons = new ArrayList<>();
+    private IntersectionDeciding intersectionDeciding;
+
+    public static AtlasEntityPolygonsFilter forConfiguration(final Configuration configuration)
```

Comment:
Is there any value in a non-configuration based constructor?  Take in a Polygon/MultiPolygon or a set of them and do the same thing?

Reply:
Check out the enum and test cases. Those have those construction methods

Reply:
Hmm looking at the test cases, it looks like all users would have to do something like this:
```
final String configurationStringFormat = "{\"filter.polygons\":{\"include.wkt\":[\"%s\"],\"exclude.wkt\":[\"%s\"]}}";

AtlasEntityPolygonsFilter.forConfiguration(new StandardConfiguration(new StringResource(String.format(configurationStringFormat, polygonWkt))));
```

If all users are going to be making the same calls, would it be possible to have this common code within the AtlasEntityPolygonsFilter class?  Or if this class implements it, we don't even need to create a configuration.

I may have missed something about the enums/test cases, but let me know!